src/material-experimental/mdc-chips/chip-icons.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Output,
+} from '@angular/core';
+import {
+  CanDisable,
+  CanDisableCtor,
+  HasTabIndex,
+  HasTabIndexCtor,
+  mixinDisabled,
+  mixinTabIndex,
+} from '@angular/material/core';
+import {FocusableCell} from './grid-focus-key-manager';
+
+
+/**
+ * Directive to add CSS classes to chip leading icon.
+ * @docs-private
+ */
+@Directive({
+  selector: 'mat-chip-avatar, [matChipAvatar]',
+  host: {
+    'class': 'mat-mdc-chip-avatar mdc-chip__icon mdc-chip__icon--leading',
+    'role': 'img'
+  }
+})
+export class MatChipAvatar {
+  constructor(private _changeDetectorRef: ChangeDetectorRef,
+              private _elementRef: ElementRef) {}
+
+  /** Sets whether the given CSS class should be applied to the leading icon. */
+  setClass(cssClass: string, active: boolean) {
+    const element = this._elementRef.nativeElement;
+    active ? element.addClass(cssClass) : element.removeClass(cssClass);
+    this._changeDetectorRef.markForCheck();
+  }
+}
+
+/**
+ * Directive to add CSS classes to and configure attributes for chip trailing icon.
+ * @docs-private
+ */
+@Directive({
+  selector: 'mat-chip-trailing-icon, [matChipTrailingIcon]',
+  host: {
+    'class': 'mat-mdc-chip-trailing-icon mdc-chip__icon mdc-chip__icon--trailing',
+    '[tabIndex]': '-1',
+    'aria-hidden': 'true',
+  }
+})
+export class MatChipTrailingIcon {
+}
+
+/**
+ * Boilerplate for applying mixins to MatChipRemove.
+ * @docs-private
+ */
+class MatChipRemoveBase extends MatChipTrailingIcon {
+  constructor(public _elementRef: ElementRef) {
+    super();
+  }
+}
+
+const _MatChipRemoveMixinBase:
+  CanDisableCtor &
+  HasTabIndexCtor &
+  typeof MatChipRemoveBase =
+    mixinTabIndex(mixinDisabled(MatChipRemoveBase));
+
+/**
+ * Directive to remove the parent chip when the trailing icon is clicked or
+ * when the ENTER key is pressed on it.
+ *
+ * Recommended for use with the Material Design "cancel" icon
+ * available at https://material.io/icons/#ic_cancel.
+ *
+ * Example:
+ *
+ *     `<mat-chip>
+ *       <mat-icon matChipRemove>cancel</mat-icon>
```

Comment:
I just noticed this example (copied from the old implementation) is wrong. The element should be a button.

Reply:
The mat-icon element should be a button? This example matches the demo, which seems to work well: https://github.com/angular/components/blob/master/src/dev-app/chips/chips-demo.html#L47

I did put 'role': 'button' on the host element for matChipRemove:
https://github.com/angular/components/pull/16384/files#diff-350c9b188f681cf2a48614ce9b6a4cffR100

Reply:
Adding the role also works, it's just a style thing. We typically tell people to do 
```html
<button mat-icon-button aria-label="...">
  <mat-icon>...</mat-icon>
</button>
```
rather than putting click handlers on icons directly; it's mostly about setting a good example

Reply:
That makes sense! The mdc classes right now don't play nicely with mat-icon-button. Can I fix this in a follow up PR?